### ISSUE TITLE
CRDs are not upgraded in 1.0+ chart

### ISSUE THREAD

**Issue by rkashasl** [aws/karpenter-provider-aws#7420]
### Description

**Observed Behavior**:
Installing kerpenter with kustomization and flux using yaml:
```
---
apiVersion: source.toolkit.fluxcd.io/v1
kind: HelmRepository
metadata:
  name: karpenter
  namespace: flux-system
spec:
  interval: 17m
  type: oci
  url: oci://public.ecr.aws/karpenter/
---
apiVersion: helm.toolkit.fluxcd.io/v2
kind: HelmRelease
metadata:
  name: karpenter
  namespace: kube-system
spec:
  install:
    createNamespace: true
    crds: Create
  upgrade:
    crds: CreateReplace
  interval: 9m
  chart:
    spec:
      chart: karpenter
      version: 1.0.8
      sourceRef:
        kind: HelmRepository
        name: karpenter
        namespace: flux-system
      interval: 14m
  values:
    priorityClassName: system-cluster-critical
    logLevel: info
    controller:
      resources:
        requests:
          cpu: 0.2
          memory: 768Mi
        limits:
          cpu: 0.2
          memory: 768Mi
    settings:
      clusterName: "${clusterName}"
      defaultInstanceProfile: "KarpenterInstanceProfile-${clusterName}"
      interruptionQueue: "KarpenterInterruptions-${clusterName}"
      featureGates:
        driftEnabled: true
    serviceAccount:
      create: false
      name: karpenter-oidc
``` 
After installation all CRDs are with old scheme. Tried to create nodepool with changed value of spec.disruption.consolidationPolicy from WhenUnderutilized to WhenEmptyOrUnderutilized leading to an error:
```
Warning  ReconciliationFailed     14m  kustomize-controller  NodePool/arm dry-run failed (Invalid): NodePool.karpenter.sh "arm" is invalid: [spec.disruption.consolidationPolicy: Unsupported value: "WhenEmptyOrUnderutilized": supported values: "WhenEmpty", "WhenUnderutilized", <nil>: Invalid value: "null": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation]
``` 

**Expected Behavior**:
CRDs should be installed or upgraded to newest one according to chart(1.0+)
**Reproduction Steps** (Please include YAML):
Just install karpenter using kustomize with file above
**Versions**:
- Chart Version: 1.0.8
- Kubernetes Version (`kubectl version`): 1.30

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by rkashasl** [comment]
@pascalrimann i tried that as well
```
---
apiVersion: source.toolkit.fluxcd.io/v1
kind: HelmRepository
metadata:
  name: karpenter
  namespace: flux-system
spec:
  interval: 17m
  type: oci
  url: oci://public.ecr.aws/karpenter/
---
apiVersion: helm.toolkit.fluxcd.io/v2
kind: HelmRelease
metadata:
  name: karpenter
  namespace: kube-system
spec:
  install:
    createNamespace: true
    crds: Skip
  upgrade:
    crds: Skip
  interval: 9m
  chart:
    spec:
      chart: karpenter
      version: 1.0.8
      # version: 0.37.6    
      sourceRef:
        kind: HelmRepository
        name: karpenter
        namespace: flux-system
      interval: 14m
  values:
    priorityClassName: system-cluster-critical
    logLevel: info
    controller:
      resources:
        requests:
          cpu: 0.2
          memory: 768Mi
        limits:
          cpu: 0.2
          memory: 768Mi
    settings:
      clusterName: "${clusterName}"
      defaultInstanceProfile: "KarpenterInstanceProfile-${clusterName}"
      interruptionQueue: "KarpenterInterruptions-${clusterName}"
      featureGates:
        driftEnabled: true
    serviceAccount:
      create: false
      name: karpenter-oidc
---
apiVersion: helm.toolkit.fluxcd.io/v2
kind: HelmRelease
metadata:
  name: karpenter-crd
  namespace: kube-system
spec:
  interval: 10m
  install:
    createNamespace: true
  upgrade:
    remediation:
      retries: 1
  chart:
    spec:
      chart: karpenter-crd
      version: "1.0.8"
      reconcileStrategy: ChartVersion      
      sourceRef:
        kind: HelmRepository
        name: karpenter
        namespace: flux-system
``` 
But i got similar result. Tried to apply a crd with changed according to changelog, but get errors that my fields not declared in schema

` Warning  ReconciliationFailed     45s                  kustomize-controller  EC2NodeClass/bottlerocket dry-run failed: failed to create typed patch object (/bottlerocket; karpenter.k8s.aws/v1beta1, Kind=EC2NodeClass): errors:  .spec.amiSelectorTerms[0].alias: field not declared in schema   

 .spec.kubelet: field not declared in schema
`
```
---
apiVersion: karpenter.k8s.aws/v1beta1
kind: EC2NodeClass
metadata:
  name: bottlerocket
spec:
  kubelet:
    evictionHard:
      memory.available: 0.5Gi
      nodefs.available: 10%
      nodefs.inodesFree: 10%
    evictionMaxPodGracePeriod: 180
    evictionSoft:
      memory.available: 800Mi
      nodefs.available: 15%
      nodefs.inodesFree: 15%
    evictionSoftGracePeriod:
      memory.available: 3m0s
      nodefs.available: 3m0s
      nodefs.inodesFree: 3m0s
    kubeReserved:
      cpu: 250m
      ephemeral-storage: 1Gi
      memory: 200Mi
    systemReserved:
      cpu: 250m
      ephemeral-storage: 1Gi
      memory: 200Mi
  amiFamily: Bottlerocket
  amiSelectorTerms:
    - alias: bottlerocket@latest  
  blockDeviceMappings:
  - deviceName: /dev/xvda
    ebs:
      deleteOnTermination: true
      encrypted: true
      volumeSize: ${rootVolumeSize:="20Gi"}
      volumeType: gp3
  - deviceName: /dev/xvdb
    ebs:
      deleteOnTermination: true
      encrypted: true
      volumeSize: ${dataVolumeSize:="50Gi"}
      volumeType: gp3
  role: KarpenterNodeRole-${clusterName}
  securityGroupSelectorTerms:
  - tags:
      aws:eks:cluster-name: ${clusterName}
  subnetSelectorTerms:
  - tags:
      kubernetes.io/role/internal-elb: "1"
      kubernetes.io/cluster/${clusterName}: "shared"
  tags:
    eks:cluster-name: ${clusterName}
    asy_monitoring_plan: ${monitoringPlan:="default"}
  metadataOptions:
    httpEndpoint: enabled
    httpPutResponseHopLimit: 2    
---
apiVersion: karpenter.k8s.aws/v1beta1
kind: EC2NodeClass
metadata:
  name: al2023
spec:
  kubelet:
    evictionHard:
      memory.available: 0.5Gi
      nodefs.available: 10%
      nodefs.inodesFree: 10%
    evictionMaxPodGracePeriod: 180
    evictionSoft:
      memory.available: 800Mi
      nodefs.available: 15%
      nodefs.inodesFree: 15%
    evictionSoftGracePeriod:
      memory.available: 3m0s
      nodefs.available: 3m0s
      nodefs.inodesFree: 3m0s
    kubeReserved:
      cpu: 250m
      ephemeral-storage: 1Gi
      memory: 200Mi
    systemReserved:
      cpu: 250m
      ephemeral-storage: 1Gi
      memory: 200Mi
  amiFamily: AL2023
  amiSelectorTerms:
    - alias: al2023@latest  
  blockDeviceMappings:
  - deviceName: /dev/xvda
    ebs:
      deleteOnTermination: true
      encrypted: true
      volumeSize: ${rootVolumeSize:="50Gi"}
      volumeType: gp3
  role: KarpenterNodeRole-${clusterName}
  securityGroupSelectorTerms:
  - tags:
      aws:eks:cluster-name: ${clusterName}
  subnetSelectorTerms:
  - tags:
      kubernetes.io/role/internal-elb: "1"
      kubernetes.io/cluster/${clusterName}: "shared"
  tags:
    eks:cluster-name: ${clusterName}
    asy_monitoring_plan: ${monitoringPlan:="default"}
  metadataOptions:
    httpEndpoint: enabled
    httpPutResponseHopLimit: 2
```

**Comment by rkashasl** [comment]
apiVersion has to be upgraded to v1 even upgrading to 1.0 not 1.1 and then it works
